include/core/truthTable/truth_table.hpp

```diff
@@ -55,6 +55,35 @@ namespace syrec {
                 }
             }
 
+            static auto findMissingCube(TruthTable::Cube::Vector const& p1SigVec) -> Cube {
```

Comment:
Could you briefly summarize what this function is supposed to do? 
It would be really helpful to add more comments to your code so that it get's easier to understand.

At the very least, it is not very scalable as it allocates huge vectors and only works up to probably something like 32 lines.

Reply:
I will add the comments to the other parts of the code in the following commit. The missing cubes function finds the a cube which is missing from the range  0 to 2^n -1 of cubes.

Reply:
Hm.. I see. But why is such a function required in the synthesis?
It really limits the scalability of the overall approach and significantly increases the memory footprint..
It should not be necessary to build a huge list of bit strings during the synthesis.

Reply:
Lets imagine we have cube on the p' edge of the dd. We want to move it to the n edge of dd. However, the the cube on the p edge is already present on the n edge. Therefore, we need to transform the cub of the p edge to the cube which is not present on the n edge (that is why I'm using missing cubes function). 

I understand it is an issue, can you suggest something else?

Reply:
A couple of observations:
 - it should be totally sufficient to return early once a suitable missing bit string has been found.
 - it is totally wasteful to allocate a vector of 64bit variables when that vector is then only used to indicate whether a certain entry is present (i.e., a Boolean value)
 - it should not be necessary at all to allocate 2**n entries
 - it should not be necessary to convert between cubes and integers
 - the loop bound in the last for loop is wrong (should iterate over all 2**n possible values)
 - the name `p1SigVec` has nothing to do with the function here

A rather simple solution (untested):
```c++
if (p1SigVec.empty()) {
  return {};
}

const auto bitwidth = p1SigVec[0].size();
std::uint64_t idx = 0U;
while(true) {
  auto cube = Cube::fromInteger(idx);
  if (p1SigVec.find(cube) == p1SigVec.end()) {
    return cube;
  }
  ++idx;
}
```
Depending on the assumptions on the incoming vector, it might be possible to make this faster. For example, if the initial vector is sorted, then one can just look for the first "gap".

Reply:
done 👍🏻